moha/rauk/rauk.py

```diff
@@ -0,0 +1,175 @@
+
+    param_nodiag_mtrx = np.zeros((n_sites, n_sites))
+    for atom1, atom2, _ in connectivity:
+        atom1_name, site1 = get_atom_type(atom1)
+        atom2_name, site2 = get_atom_type(atom2)
+        key1 = atom1_name #if atom1_coord else atom1_name
+        key2 = atom2_name #if atom2_coord else atom2_name
+
+        #print((site1 - 1, site2 - 1), key1 + key2, self.bond_dictionary[key1 + key2])
+        param_nodiag_mtrx[site1 - 1, site2 - 1] = bond_dictionary[key1 + key2]
+        
+    param_nodiag_mtrx = np.minimum(param_nodiag_mtrx, param_nodiag_mtrx.T)
+    param_nodiag_mtrx = param_nodiag_mtrx
+    param_diag_mtrx = csr_matrix(param_diag_mtrx)
+    param_nodiag_mtrx = csr_matrix(param_nodiag_mtrx)
+    
+    one_body = param_diag_mtrx + param_nodiag_mtrx
+    
+    return one_body
+
+def assign_rauk_parameters(connectivity, atom_types, atoms_num, n_sites):
+    r"""Assigns the alpha and beta value from Rauk's table in matrix form"""
+
+    dir_path = os.path.dirname(os.path.abspath(__file__))
+
+    # Paths to the JSON files
+    hx_dict_path = os.path.join(dir_path, 'hx_dictionary.json')
+    kxy_matrix_path = os.path.join(dir_path, 'kxy_matrix_1.json')
+
+    # Load the dictionary from the hx_dictionary.json file
+    with open(hx_dict_path, 'r') as file:
+        hx_dictionary = json.load(file)
+
+    # Load the matrix from the kxy_matrix_1.json file
+    with open(kxy_matrix_path, 'r') as file:
+        kxy_matrix_1_list = json.load(file)
+
+    # Convert list back to numpy array
+    kxy_matrix_1 = np.array(kxy_matrix_1_list)
+    kxy_matrix = np.minimum(kxy_matrix_1, kxy_matrix_1.T) #Symmetric matrix
+
+    alphaC = -0.414 #Value for sp2 orbital of Carbon atom.
+    betaC = -0.0533 #Value for sp2 orbitals of Carbon atom.
+    
+    
+    # Create atom dictionary using predefined values without overlap parameters
+    atom_dictionary = {}
+    atom_types.pop(0)
+    print(atom_types)
+    for atom in atom_types:
+        atom_dictionary[atom] = alphaC + hx_dictionary[atom] * abs(betaC)
+ 
+    #print(atom_dictionary)
+    # Create bond dictionary using predefined values without overlap parameters
+    bond_dictionary = {}
+    for i, atom in enumerate(atom_types):
+        next_atom = atom_types[i + 1] if i < len(atom_types) - 1 else atom_types[0]
+        #print(atom, next_atom)
+        index1 = list(hx_dictionary.keys()).index(atom)
+        index2 = list(hx_dictionary.keys()).index(next_atom)
+        bond_key = atom + next_atom
+        bond_value = kxy_matrix[index1, index2] * abs(betaC)
+        bond_dictionary[bond_key] = bond_value
+        #print(bond_key, bond_value)
+        bond_dictionary[next_atom + atom] = bond_value  # Ensure symmetry
+    #print(bond_dictionary)
+    
+    one_body = build_one_body_matrix(connectivity, atom_dictionary, atoms_num, n_sites, bond_dictionary)
+    return one_body
+
+
+def compute_param_dist_overlap(connectivity, atom_types, atoms_num, n_sites, dist_atoms):
```

Comment:
All of the functions from this line and below, should go into a separate file called `ppp_dict.py`